### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by InsaneAwesomeTony** [wojtryb/Shortcut-Composer#56] [Ideas]
Hey, big fan of this addon
I would have added this myself if I knew how to code and not break everything in a spectacular ball of flame kinda way

The option to add more brush sets to it
3 might be enough for normal people, i sadly have anxiety on choosing my 3 brush sets and would like to have two more be added. If there was a button to add an additional color (red, green, blue) so more can be added, it would be pretty neat

Thank you for reading this.

**Comment by wojtryb** [comment]
Hi,

I'd love to have it, but krita (for various reasons) does not allow to create new actions from the plugin itself - everything you see in shortcut settings must be filled by the human.

Unluckliy, current version of the plugin (1.1.1) does not allow to easily add new preset tags. Your options are:

1. Downgrade plugin to 1.0.1 (On github switch from `main` to `1.0.1`, download and install the plugin)
2. Wait for 1.2.0 (probably weeks)
3. Upgrade to development version of 1.2.0 (Switch from `main` to `feature/pie-specific-settings`)

If you decide to do any of these, you can follow this tutorial (which won't work on 1.1.1):
1. Download and install the plugin of specific version
2. From krita: `Settings > Manage Resources > Open Resource Folder > pykrita > shortcut_composer`
3. Edit `actions.py` file. Add your two new shortcuts under existing blue one
```python
    # Use pie menu to pick one of presets from tag specified in settings.
    # Set tool to FREEHAND BRUSH if current tool does not allow to paint
    templates.PieMenu(
        name="Pick brush presets (blue)",
        controller=controllers.PresetController(),
        instructions=[instructions.SetBrushOnNonPaintable()],
        values=Tag(Config.TAG_BLUE.read()),
        background_color=QColor(70, 70, 105, 190),
        active_color=QColor(110, 160, 235),
    ),

    templates.PieMenu(
        name="Pick brush presets (color1)",
        controller=controllers.PresetController(),
        instructions=[instructions.SetBrushOnNonPaintable()],
        values=Tag("tag_1_name"),
        background_color=QColor(70, 70, 105, 190),  # Red, gree, blue, opacity 0-255
        active_color=QColor(110, 160, 235),  # no transparency allowed
    ),

    templates.PieMenu(
        name="Pick brush presets (color2)",
        controller=controllers.PresetController(),
        instructions=[instructions.SetBrushOnNonPaintable()],
        values=Tag("tag_2_name"),
        background_color=QColor(70, 70, 105, 190),
        active_color=QColor(110, 160, 235),
    ),

    # .......................................
    # Insert your actions implementation here
    # .......................................

]
```
4. Edit `actions.action` file. Add your two new actions under the blue one again. Make sure the names are exactly the same as before.
```xml
        <Action name="Pick brush presets (blue)">
        <activationFlags>1</activationFlags>
        </Action>

        <Action name="Pick brush presets (color1)">
        <activationFlags>1</activationFlags>
        </Action>

        <Action name="Pick brush presets (color2)">
        <activationFlags>1</activationFlags>
        </Action>

    </Actions>
    <Actions category="Scripts">
        <text>Shortcut Composer: Utilities</text>
```

I won't be adding more tags to the official plugin, as I believe 3 are enough for most users, so you have to do it on your own :)

Cheers

**Comment by InsaneAwesomeTony** [comment]
thank you for the help! I will try to do this on my own with the help you have provided! :D
